src/marks/Link.js

```diff
@@ -63,30 +68,61 @@ const Link = TipTapLink.extend({
 		return ['a', {
 			...mark.attrs,
 			href: domHref(mark, this.options.relativePath),
+			'data-md-href': mark.attrs.href,
 			rel: 'noopener noreferrer nofollow',
 		}, 0]
 	},
 
 	addProseMirrorPlugins() {
 		const plugins = this.parent()
-			// remove original handle click
+			// remove upstream link click handle plugin
 			.filter(({ key }) => {
 				return !key.startsWith('handleClickLink')
 			})
 
-		if (!this.options.openOnClick) {
-			return plugins
-		}
-
-		// add custom click handler
+		// Custom click handler plugins
 		return [
 			...plugins,
-			clickHandler({
-				editor: this.editor,
-				type: this.type,
-				onClick: this.options.onClick,
+			new Plugin({
+				key: new PluginKey('textHandleClickLink'),
+				props: {
+					handleDOMEvents: {
+						// Prevent paste on middle click in links (Linux only)
+						// Open link in new tab on middle click
+						pointerup: (view, event) => {
+							if (event.target.closest('a') && event.button === 1 && !event.ctrlKey && !event.metaKey && !event.shiftKey) {
+								this.storage.nopaste = true
+								event.preventDefault()
+
+								const linkElement = event.target.closest('a')
+								window.open(linkElement.href, '_blank')
+							}
+						},
+						// Prevent paste on middle click in links (Linux only)
+						paste: (view, event) => {
+							if (event.target.closest('a') && this.storage.nopaste) {
+								event.stopPropagation()
+								event.preventDefault()
+								event.stopImmediatePropagation()
+								this.storage.nopaste = false
+							}
+						},
```

Comment:
So the `preventDefault()` in `pointerup` is not enough?

That's a pitty as this seems brittle. For example what would happen if i use middle click to open a link in a new tab on windows and then later on try to paste something into a link? Wouldn't `this.storage.nopaste` still be true?

Reply:
Unfortunately the `preventDefault()` in `pointerup` is not sufficient according to my tests. But you have a fair point. Mhh, have to think more about this.

Reply:
The best option that comes to my mind is to set `this.storage.nopaste = true` only on Linux (by checking the user agent). What do you think @max-nextcloud?

Reply:
I implemented this now: https://github.com/nextcloud/text/pull/5158/commits/d27d2fa91e421f95cff0abf7f43b5266f4b70b45#diff-90d862ff7cd51c7b18035dbb20c02bc177926bf7224193d0a58b5f870f7a8adbR96-R98